src/psyclone/tests/psyir/nodes/node_test.py

```diff
@@ -80,16 +81,42 @@ def test_node_coloured_name():
         tnode.node_str()
     assert ("_text_name is an abstract attribute which needs to be given a "
             "string value in the concrete class 'Node'." in str(err.value))
-
-    # Check that we can change the name of the Node and the colour associated
-    # with it
+    # Exception as _colour has not been set
     tnode._text_name = "ATest"
+    with pytest.raises(NotImplementedError) as err:
+        _ = tnode.coloured_name()
+    assert ("The _colour attribute is abstract so needs to be given a string "
+            "value in the concrete class 'Node'." in str(err.value))
+    # Valid values
     tnode._colour = "white"
     assert tnode.coloured_name(False) == "ATest"
     assert tnode.coloured_name(True) == colored("ATest", "white")
 
 
+def test_node_coloured_name_exception(monkeypatch):
+    '''Test that the expected exception is raised if the colour provided
+    to the colored function is invalid. Note, an exception if the
+    termcolor package is installed. Therefore we monkeypatch the
+    function to force the exception whether termcolor is installed or
+    not.
+
+    '''
+    def dummy(error):
+        '''Utility used to raise the required error from a lambda function.'''
+        raise error()
+
+    monkeypatch.setattr(node, "colored", lambda _1, _2: dummy(KeyError))
```

Comment:
I don't think you need to pass in the error class or use a `lambda`:

    def dummy(_1, _2):
        raise KeyError()

    monkeypatch.setattr(node, "colored", dummy)